tests/testthat/test-implicit_assignment_linter.R

```diff
@@ -317,11 +304,27 @@ test_that("implicit_assignment_linter blocks disallowed usages in function calls
       return(x <- x + 1)
     }"),
     list(
-      list(message = lint_message, line_number = 2L, column_number = 9L),
-      list(message = lint_message, line_number = 3L, column_number = 12L)
+      list(message = lint_message, line_number = 2L, column_number = 7L),
+      list(message = lint_message, line_number = 3L, column_number = 10L)
     ),
     linter
   )
+
+  expect_lint(
+    trim_some("
+      map(
+        .x = 1:4,
+        .f = ~ .x + 1 -> x
+      )"),
+    lint_message,
+    linter
+  )
```

Comment:
Can you please also add a test with a version of this that does _not_ lint?

Reply:
see here: https://github.com/r-lib/lintr/issues/2138#issuecomment-1714362558

in this PR we could do f. = ~(.x + 1 -> x)

but the last PR invalidates that I'm favor of requiring braces

Reply:
Okay, then let's leave that for the later PRs.